src/main/java/org/mule/tools/devkit/sonar/checks/pom/DistributionManagementByCategoryCheck.java

```diff
@@ -0,0 +1,86 @@
+            } else if (snapshotRepository == null) {
+                logAndRaiseIssue(issues, "Distribution Management is missing required <snapshotRepository> configuration.");
+            } else {
+                switch (category.toUpperCase()) {
+                    case "PREMIUM":
+                    case "STANDARD":
+                    case "SELECT":
+                    case "CERTIFIED":
+                        checkPremiumOrSelectOrCertified(category, issues, deploymentRepository, snapshotRepository);
+                        break;
+
+                    case "COMMUNITY":
+                        checkCommunity(category, issues, deploymentRepository, snapshotRepository);
+                        break;
+
+                    default:
+                        logAndRaiseIssue(issues, "Invalid category specified in pom.xml");
+                        break;
+                }
+            }
+        }
+
+        return issues;
+    }
+
+    private void checkCommunity(String category, @NonNull List<PomIssue> issues, @NonNull DeploymentRepository deploymentRepository,
+            @NonNull DeploymentRepository snapshotRepository) {
+        if (!(deploymentRepository.getId().equals("mulesoft-releases") && deploymentRepository.getName().equals("MuleSoft Repository")
+                && deploymentRepository.getUrl().equals("http://repository-master.mulesoft.org/releases/") && snapshotRepository.getId().equals("mulesoft-snapshots")
+                && snapshotRepository.getName().equals("MuleSoft Snapshot Repository") && snapshotRepository.getUrl().equals("http://repository-master.mulesoft.org/snapshots/") && !snapshotRepository
+                    .isUniqueVersion())) {
+            logAndRaiseIssue(issues, String.format("Distribution Management must be properly configured in pom.xml for %s category.", category));
+        }
+
+    }
+
+    private void checkPremiumOrSelectOrCertified(String category, @NonNull List<PomIssue> issues, @NonNull DeploymentRepository deploymentRepository,
+            @NonNull DeploymentRepository snapshotRepository) {
+        if (!(deploymentRepository.getId().equals("mule-ee-releases") && deploymentRepository.getName().equals("MuleEE Releases Repository")
+                && deploymentRepository.getUrl().equals("https://repository-master.mulesoft.org/nexus/content/repositories/releases-ee/")
+                && snapshotRepository.getId().equals("mule-ee-snapshots") && snapshotRepository.getName().equals("MuleEE Snapshots Repository")
+                && snapshotRepository.getUrl().equals("https://repository-master.mulesoft.org/snapshots/") && !snapshotRepository.isUniqueVersion())) {
+            logAndRaiseIssue(issues, String.format("Distribution Management must be properly configured in pom.xml for %s category.", category));
+        }
+
+    }
+
+    private final void logAndRaiseIssue(List<PomIssue> issues, String message) {
+        LoggerFactory.getLogger(getClass()).info(message);
```

Comment:
There's no need to log here, logging is done in MavenSensor class. Only responsibility of this class is to collect the PomIssues and return them in an Iterable. So I guess this method should be gone and only call `issues.add()` in those places.